// bbb/lib/bbb.js
/*!
 * bbb
 * Copyright(c) 2011 Etienne Lachance <et@etiennelachance.com>
 * MIT Licensed
 */

var bbb = Object();

require('joose')
require('joosex-namespace-depended')
require('hash')

exports.url = "";
exports.securitySalt = "";

/**
 * crypt
**/
exports.crypt = function(securitySalt, cmd, params) {
    return Hash.sha1(cmd+params+securitySalt);
}

exports.checksum = function(securitySalt, cmd, params) {
    return this.crypt(securitySalt, cmd, params);
}

exports.generateQueryString = function(clientId, securitySalt, cmd, params) {
    var checksum = this.checksum(securitySalt, cmd, params);
    var queryString = "/"+clientId+"/api/"+cmd+"?"+params+"&checksum="+checksum;
    return queryString;
}
/**
 * query
**/
exports.query = function(cmd, params, callback) {
    var http = require('http');
    var xml2js = require('xml2js');
    var sys = require('sys');

    var queryString = this.generateQueryString("bigbluebutton", this.securitySalt, cmd, params);
    console.log("==== BigBlueButton Query String : "+queryString);

    var options = {
      host: this.url,
      port: 80,
      path: queryString
    };

    var parser = new xml2js.Parser();
    parser.addListener('end', function(result) {
        callback(result);
    });

    request = http.get(options, function(res) {}).on('error', function(e) {
        console.log("Got error: " + e.message);
    });

    request.on('response', function (response) {
      response.on('data', function (chunk) {
        parser.parseString(chunk);
      });
    });
}
